app/app.go

```diff
@@ -154,6 +155,91 @@ func (a *App) getFirstServerRunTimestamp() (int64, *model.AppError) {
 	return value, nil
 }
 
+// WarnMetricPrefix represent the prefix used for any warn metric type
+const WarnMetricPrefix = "warn_metric_"
+
+func (a *App) GetWarnMetricsStatus() (map[string]bool, *model.AppError) {
+	systemDataList, appErr := a.Srv().Store.System().Get()
+	if appErr != nil {
+		return nil, appErr
+	}
+
+	result := map[string]bool{}
+	for key, value := range systemDataList {
+		if strings.HasPrefix(key, WarnMetricPrefix) {
+			result[key] = value == "true"
+		}
+	}
+
+	return result, nil
+}
+
+func (a *App) SetWarnMetricStatus(warnMetricId string) *model.AppError {
+	mlog.Info("Storing user acknowledgement for warn metric", mlog.String("metric", warnMetricId))
+	if err := a.Srv().Store.System().SaveOrUpdate(&model.System{
+		Name:  warnMetricId,
+		Value: "ack",
+	}); err != nil {
+		mlog.Error("Unable to write to database.", mlog.Err(err))
+		return model.NewAppError("SetWarnMetricStatus", "app.system.number_active_users_warn_metric.store.app_error", nil, err.Error(), http.StatusInternalServerError)
+	}
+	return nil
+}
+
+func (a *App) NotifyAdminsOfWarnMetricStatus(warnMetricId, warnMetricMessage string) *model.AppError {
+	perPage := 25
+	userOptions := &model.UserGetOptions{
+		Page:     0,
+		PerPage:  perPage,
+		Role:     model.SYSTEM_ADMIN_ROLE_ID,
+		Inactive: false,
+	}
+
+	// get sysadmins
+	var sysAdmins []*model.User
+	for {
+		sysAdminsList, err := a.GetUsers(userOptions)
+		if err != nil {
+			mlog.Error("Cannot obtain list of system admins!")
+			return err
+		}
+
+		sysAdmins = append(sysAdmins, sysAdminsList...)
+
+		if len(sysAdminsList) < perPage {
+			mlog.Debug("Number of system admins is less than limit", mlog.Int("admin count", len(sysAdminsList)))
+			break
+		}
+	}
+
+	for _, sysAdmin := range sysAdmins {
+		channel, appErr := a.GetOrCreateDirectChannel(sysAdmin.Id, sysAdmin.Id)
+		if appErr != nil {
+			mlog.Error("Cannot create channel for system notifications!", mlog.String("Admin Id", sysAdmin.Id))
+			return appErr
+		}
+
+		post := &model.Post{
+			UserId:    sysAdmin.Id,
```

Comment:
Here should be the `systemBot` created for this kind of tasks

Reply:
see above.